### PR TITLE
(PCP-836) Add powershell and shell features

### PR DESCRIPTION
**Pull request by MikaelSmith**
Add the shell feature on non-Windows hosts, and add the powershell
feature if a powershell executable is found.

### CODE REVIEW

File: lib/src/modules/task.cc [hunk 1]
```diff
@@ -174,6 +180,13 @@ Task::Task(const fs::path& exec_prefix,
     client_.set_client_cert(crt, key);
     client_.set_supported_protocols(CURLPROTO_HTTPS);
     client_.set_proxy(proxy);
+
+#ifndef _WIN32
+    features_.insert("shell");
+#endif
+    if (!lth_exec::which("powershell").empty()) {
```
Comment (nicklewis) [inline lib/src/modules/task.cc]: Do powershell tasks work on non-windows systems? I'm unclear whether the "built-in task interpreter" for ps1 on windows does anything that would be necessary elsewhere. Specifically, it passes `-NoProfile -NonInteractive -NoLogo -ExecutionPolicy Bypass`.

Comment (MikaelSmith) [reply lib/src/modules/task.cc]: I'm not sure those arguments are needed with powershell on other platforms. They're present in the powershell 6 beta, but that might just be so scripts don't break.

Comment (MikaelSmith) [reply lib/src/modules/task.cc]: Hard-coded it so powershell is only matched on Windows.

**Comment by MikaelSmith** [comment]
jenkins please test this

**Comment by MikaelSmith** [comment]
Acceptance tests are still running off the 2nd commit, and I expect to be sufficient to validate the changes.

**Comment by MikaelSmith** [comment]
Everything's green @nicklewis .

**Pull request merged**